libcloud/compute/drivers/packet.py

```diff
@@ -88,11 +98,88 @@ class PacketNodeDriver(NodeDriver):
                       'failed': NodeState.ERROR,
                       'active': NodeState.RUNNING}
 
-    def list_nodes(self, ex_project_id):
-        data = self.connection.request('/projects/%s/devices' %
-                                       (ex_project_id),
-                                       params={'include': 'plan'}
-                                       ).object['devices']
+    def __init__(self, key, project=None):
+        # initialize a NodeDriver for Packet using the API token
+        # and optionally the project (name or id)
+        # If project specified we need to be sure this is a valid project
+        # so we create the variable self.project_id
+        super(PacketNodeDriver, self).__init__(key=key, project=None)
```

Comment:
I assume you don't actually want to pass ``project`` keyword argument to the parent class (``BaseDriver``) constructor, since the parent constructor doesn't support that argument (it will simply ignore it, but it may confuse people reading the code).

Reply:
Fixed